javascript/ql/src/Statements/IgnoreArrayResult.ql

```diff
@@ -0,0 +1,47 @@
+/**
+ * @name Ignoring result from pure array method
+ * @description The array methods do not modify the array, ignoring the result of such a call is therefore generally an error.
+ * @kind problem
+ * @problem.severity warning
+ * @id js/ignore-array-result
+ * @tags maintainability,
+ *       correctness
+ * @precision high
+ */
+
+import javascript
+import Expressions.ExprHasNoEffect
+
+DataFlow::SourceNode callsArray(DataFlow::TypeBackTracker t, DataFlow::MethodCallNode call) {
+  isIgnoredPureArrayCall(call) and
```

Comment:
It may be better to distribute this into the first disjunct. It is implied by the recursive call in the second disjunct.

Reply:
It definitely makes it more readable, as it now follows the usual type-tracking pattern.